meerkat_abacus/data_management.py

```diff
@@ -895,10 +908,16 @@ def new_data_to_codes(engine=None, debug_enabled=True, restrict_uuids=None,
     session.commit()
 
     for data_type in data_types.data_types(param_config=param_config):
-        # TODO: this piece of performance improvement broke "ongoing" status of alerts
-        # Removing untill further investigation.
-        # if only_forms and data_type["form"] not in only_forms:
-        #     continue
+        if data_type["form"] not in restrict_uuids:
+            link_forms = []
+            for link in links_by_type.get(data_type["name"], []):
+                link_forms += [link["to_form"], link["from_form"]]
+            found = False
+            for l in set(link_forms):
+                if l in restrict_uuids.keys():
+                    found = True
+            if found is False:
+                continue
```

Comment:
Why do you create a set from link_forms?
`if found is False:` looks evil. I'd go with `if not found:` or even better sth like this:
```python
if any([link not in restrict_uuids.keys() for link in set(link_forms)]):
  continue
```

Reply:
just creating a set to remove any duplicates in the list